core/core.go

```diff
@@ -458,12 +464,19 @@ func downloadBazelToCAS(version string, bazeliskHome string, repos *Repositories
 	f.Close()
 	actualSha256 := strings.ToLower(fmt.Sprintf("%x", h.Sum(nil)))
 
-	pathToBazelInCAS := filepath.Join(casDir, actualSha256, "bin", "bazel"+platforms.DetermineExecutableFilenameSuffix())
-	if err := os.MkdirAll(filepath.Dir(pathToBazelInCAS), 0755); err != nil {
+	bazelInCASBasename := "bazel" + platforms.DetermineExecutableFilenameSuffix()
+	pathToBazelInCAS := filepath.Join(casDir, actualSha256, "bin", bazelInCASBasename)
+	dirForBazelInCAS := filepath.Dir(pathToBazelInCAS)
+	if err := os.MkdirAll(dirForBazelInCAS, 0755); err != nil {
 		return "", "", fmt.Errorf("failed to MkdirAll parent of %s: %w", pathToBazelInCAS, err)
 	}
 
-	tmpPathInCorrectDirectory := pathToBazelInCAS + ".tmp"
+	tmpPathFile, err := os.CreateTemp(dirForBazelInCAS, bazelInCASBasename+".tmp")
```

Comment:
I believe you must `tmpPathFile.Close()` right away. On Windows, having `tmpPathFile` still be open would prevent the `os.Rename` later from succeeding.

Reply:
Having to remember that Windows exists is the worst.

Reply:
```
2024/09/30 18:32:03 could not download Bazel: failed to download bazel: failed to move C:\b\u5s5am3n\execroot\_main\_tmp\ab7edd0c907fc9425929721be19c9e77\home.5X2COn\downloads\_tmp\476b775359a16b711435ca9932ca2e33b41dab97c5d89d1634d66b1326233427 to C:\b\u5s5am3n\execroot\_main\_tmp\ab7edd0c907fc9425929721be19c9e77\home.5X2COn\downloads\sha256\bf4adcd2764dbca6332092950f41c5ae435a38032c1c00a5558ef963b32107a1\bin\bazel.exe.tmp3033836440: rename C:\b\u5s5am3n\execroot\_main\_tmp\ab7edd0c907fc9425929721be19c9e77\home.5X2COn\downloads\_tmp\476b775359a16b711435ca9932ca2e33b41dab97c5d89d1634d66b1326233427 C:\b\u5s5am3n\execroot\_main\_tmp\ab7edd0c907fc9425929721be19c9e77\home.5X2COn\downloads\sha256\bf4adcd2764dbca6332092950f41c5ae435a38032c1c00a5558ef963b32107a1\bin\bazel.exe.tmp3033836440: Access is denied.
```